server/container/guice/cassandra-guice/src/test/java/org/apache/james/DockerCassandraRule.java

```diff
@@ -55,8 +57,13 @@ public void await() {
 
     @Override
     public Module getModule() {
-        String keyspace = new RandomStringGenerator.Builder().withinRange('a', 'z').build().generate(12);
-        return (binder) -> binder.bind(CassandraSessionConfiguration.class).toInstance(() -> getCassandraConfigurationForDocker(keyspace));
+        lastKeyspace = new RandomStringGenerator.Builder().withinRange('a', 'z').build().generate(12);
+        return (binder) -> binder.bind(CassandraSessionConfiguration.class).toInstance(() -> getCassandraConfigurationForDocker(lastKeyspace));
+    }
+
+    public String getLastKeyspace() {
+        Preconditions.checkNotNull(lastKeyspace);
```

Comment:
Preconditions in a get. That seems weird to me.

Reply:
Why? To get the last keyspace I need to ensure that we got one....

I prefer this than returning an optional and let the caller handling the invalid state.

Reply:
Then it's a checkState, not a checkNotNull